### PR TITLE
[RabbitMQ] Add queue config to service config

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose
fixes: https://github.com/ballerina-platform/ballerina-library/issues/6629

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/tests/rabbitmq_client_tests.bal [hunk 1]
```diff
@@ -1164,6 +1214,62 @@ service object {
...
+service object {
+    remote function onMessage(BytesMessage message) {
+        string|error messageContent = 'string:fromBytes(message.content);
+        if messageContent is string {
+            queueConfigNew = messageContent;
+            log:printInfo("The reply message received: " + messageContent);
+        } else {
+            log:printError("Error occurred while retrieving the message content.", 'error = messageContent);
+        }
+    }
```
Comment (ayeshLK) [inline ballerina/tests/rabbitmq_client_tests.bal:1235]:
```suggestion
    remote function onMessage(BytesMessage message) returns error? {
        string messageContent = check 'string:fromBytes(message.content);
        queueConfigNew = messageContent;
        log:printInfo("The reply message received: " + messageContent);
    }
```
Can't we do this ?

File: native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java [hunk 2]
```diff
@@ -167,16 +169,38 @@ public static Object detach(Environment environment, BObject listenerBObject, BO
...
+                        + RabbitMQConstants.SERVICE_CONFIG));
         String queueName;
         if (service.getNativeData(RabbitMQConstants.QUEUE_NAME.getValue()) != null) {
+            // Queue name is given as the service name
             queueName = (String) service.getNativeData(RabbitMQConstants.QUEUE_NAME.getValue());
         } else {
+            // Queue name and config given in the service config
             queueName = serviceConfig.getStringValue(RabbitMQConstants.QUEUE_NAME).getValue();
+            Map<String, Object> argumentsMap = new HashMap<>();
+            boolean durable = false;
```
Comment (ayeshLK) [inline native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java:183]: What happens if the user has defined a service-name and `queueName` in the `rabbitmq:ServiceConfig` annotation ?

File: native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java [hunk 3]
```diff
@@ -167,16 +169,38 @@ public static Object detach(Environment environment, BObject listenerBObject, BO
...
+
+            if ((BMap<BString, Object>) serviceConfig.getMapValue(RabbitMQConstants.QUEUE_CONFIG) != null) {
+
+                @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+                BMap<BString, Object> queueConfig =
+                        (BMap<BString, Object>) serviceConfig.getMapValue(RabbitMQConstants.QUEUE_CONFIG);
+                durable = queueConfig.getBooleanValue(RabbitMQConstants.QUEUE_DURABLE);
+                exclusive = queueConfig.getBooleanValue(RabbitMQConstants.QUEUE_EXCLUSIVE);
+                autoDelete = queueConfig.getBooleanValue(RabbitMQConstants.QUEUE_AUTO_DELETE);
+                if (queueConfig.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS) != null) {
```
Comment (ayeshLK) [inline native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java:195]:
```suggestion
                if (queueConfig.containsKey(RabbitMQConstants.QUEUE_ARGUMENTS)) {
```

Would this work here ?

File: native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java (hunk 2 above)
Comment (aashikam) [reply native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java]: Check this one WDYT db6ca785cc56b48625ed3d518c5b27c94ee088d0

File: native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java (hunk 3 above)
Comment (aashikam) [reply native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java]: i think the default value is null so the key will be there with null as value.

File: native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java (hunk 2 above)
Comment (aashikam) [reply native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java]: basically config replaced service name

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Review by ayeshLK — APPROVED** [review]
LGTM

**Pull request merged**